File: Tests/RegexAdapterTests.cs
using System;
using System.Text.RegularExpressions;
using Stringier.Patterns;
using static Stringier.Patterns.Pattern;
using Xunit;

namespace Tests {
	public class RegexAdapterTests {
		[Fact]
		public void Converter() {
			_ = new Regex("^hello").AsPattern();
			_ = new Regex("^hello$").AsPattern();
			Assert.Throws<PatternConstructionException>(() => new Regex("hello").AsPattern());
			Assert.Throws<PatternConstructionException>(() => new Regex("hello$").AsPattern());
		}

		[Fact]
		public void Consume() {
			Pattern pattern = new Regex("^hello").AsPattern();
			ResultAssert.Captures("hello", pattern.Consume("hello"));
			ResultAssert.Captures("hello", pattern.Consume("hello world"));
			ResultAssert.Fails(pattern.Consume("hel"));
		}
	}
}

File: Tests.FSharp/RegexAdapterTests.fs
namespace Patterns

open System
open System.Text.RegularExpressions
open Stringier.Patterns
open Defender
open Xunit

type RegexAdapterTests() =
    inherit Trial()

    [<Fact>]
    member _.``converter`` () =
        let mutable pattern = (fun () -> Regex(@"hello").AsPattern() |> ignore)
        Assert.Throws<PatternConstructionException>(Action pattern) |> ignore
        Regex(@"^hello").AsPattern() |> ignore
        pattern <- (fun () -> Regex(@"hello$").AsPattern() |> ignore)
        Assert.Throws<PatternConstructionException>(Action pattern) |> ignore
        Regex(@"^hello$").AsPattern() |> ignore
        
    [<Fact>]
    member _.``consume`` () =
        p(Regex(@"^hello"))
        |> consumes "hello" "hello"
        |> consumes "hello" "hello world"
        |> fails "hel"
        |> ignore